### ISSUE TITLE
Error on executing GluonVM sample app

### ISSUE THREAD

**Issue by bgmf** [javafxports/javafxmobile-plugin#16]
Before attempting to build our newest app with the help of the GluonVM, I tried to build the sample app:
https://github.com/gluonhq/gluon-samples/tree/master/singleview-gluonvm

Following the instructions (especially the one for the stack size), I finally faced the issue, that the Gradle task `launchIOSDevice` failed with the following exception:

```
CODESIGN DONE
UploadInternal start
START WALKING filetree in uploadInternal
DONE WALKING filetree in uploadInternal
uploadInternal will now free client with pointer jnr.ffi.provider.BoundedMemoryIO[address=0x1341b5038 size=8]
uploadInternal did now free client with pointer jnr.ffi.provider.BoundedMemoryIO[address=0x1341b5038 size=8]
uploadInternal will now free lockdownpointer jnr.ffi.provider.BoundedMemoryIO[address=0x1341b5028 size=8]
uploadInternal did now free lockdownpointer jnr.ffi.provider.BoundedMemoryIO[address=0x1341b5028 size=8]
uploadInternal done
sleep now
free instProxyClientPointer at jnr.ffi.provider.BoundedMemoryIO[address=0x1342e9a48 size=8]
freed instProxyClientPointer at jnr.ffi.provider.BoundedMemoryIO[address=0x1342e9a48 size=8]
free lockdownclientpointer at jnr.ffi.provider.BoundedMemoryIO[address=0x1342e9a38 size=8]
freed lockdownclientpointer at jnr.ffi.provider.BoundedMemoryIO[address=0x1342e9a38 size=8]
[09:21:32:485] BosonAppBuilder: Path not found, exit now 
java.lang.RuntimeException: Path not found, exit now
        at com.gluonhq.gvmbuild.LoggingSupport.logSevere(LoggingSupport.java:106)
        at com.gluonhq.gvmbuild.BosonAppBuilder.getAppPath(BosonAppBuilder.java:1015)
        at com.gluonhq.gvmbuild.BosonAppBuilder.launchInternal(BosonAppBuilder.java:954)
        at com.gluonhq.gvmbuild.BosonAppBuilder.launch(BosonAppBuilder.java:808)
        at com.gluonhq.gvmbuild.BosonAppBuilder.launchOnDevice(BosonAppBuilder.java:503)
        at org.javafxports.jfxmobile.plugin.Gvm.build(Gvm.java:113)
        at org.javafxports.jfxmobile.plugin.ios.task.IosDevice.action(IosDevice.java:42)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:784)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:751)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:121)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:60)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:626)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:581)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.base/java.lang.Thread.run(Thread.java:844)​
```

It should be noted, that the app runs (although slowly) on the simulator.

@johanvos suggested, to create an issue about this.

Daniel

**Comment by johanvos** [comment]
Sometimes there are issues indeed with launching automatically indeed.
From the log, it looks like the app was installed on the device though. Can you check that? And does it run if you just start it as a regular app by clicking it on the device?

**Comment by bgmf** [comment]
No, it wasn't. But I just realized, that I called it from the parent directory. 

When I execute gradle from the app directory and try to do the `ios` or `installIos` tasks, it finishes in 1s or so, but there is no build in the `build/javafxports/ios`(ios is completely missing).

I got some help from a more IOS affine colleague of mine, and we saw, that after the message from above, there is an `.app` file in the direcory `build/javafxports/gvm` - we now try to convert this into an installable `.ipa` file.

**Comment by bgmf** [comment]
We finally managed to package the `.app` file into a functional `.ipa` file and run it on a iPhone (sidenote: the old iPhone 4S seems to be out of scope now... :smile:).

There indeed seems to be a performance gain (the startup time of the framework, I guess - there is a short gap, a black screen, between the IOS splash image and the app main screen) which seems to be slightly shorter than in the next real-world app I'm currently working on.

So: The Gradle task still is broken, it's not able to create the `.ipa` file, but if you're willing to can create a running app with the `.app` file...

**Comment by johanvos** [comment]
Thanks for your testing work!

creating an IPA is indeed not yet supported, but 
./gradlew launchIosDevice
should work.
So I will leave this ticket open until we feel this is fixed.

**Comment by johanvos** [comment]
Can you try this again with a new project? If you create a new project with the IDE plugin, you'll see it will use the latest version of the jfxmobile plugin (2.0.5).
You can also manually change that version in an existing project.

**Comment by bgmf** [comment]
Thanks for the feedback: Will try on Monday. ok?

**Comment by johanvos** [comment]
sure :)

**Comment by bgmf** [comment]
Sorry, took me a bit to long to finally do it...

It fails on a different place, this is an excerpt of the messages printed to the console:
```
> Task :singleview-gluonvmApp:launchIOSDevice 
Setting gloal LogLevel to INFO
Configure logging for GVM tooling... BUNDLE ID = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM
[HIGGS] Creating higgs...
[13:24:02:414] com.gluonhq.higgs.Higgs: [HIGGS] mainClassName = com/gluonhq/samples/singleviewgvm/SingleViewWithGluonVM 
[13:24:02:424] com.gluonhq.higgs.Higgs: [HIGGS] arch = arm64 
missing file /Users/bgmf/Programming/Work/gluon-samples/singleview-gluonvm/singleview-gluonvmApp/build/classes/java/ios
[13:24:03:377] com.gluonhq.higgs.Higgs: Still compiling... 3551 classes left. 
[13:24:13:381] com.gluonhq.higgs.Higgs: Still compiling... 1060 classes left. 
[13:24:23:383] com.gluonhq.higgs.Higgs: Still compiling... 1808 classes left. 
[13:24:33:385] com.gluonhq.higgs.Higgs: Still compiling... 418 classes left. 
[13:24:37:431] com.gluonhq.higgs.Compiler: Compiled 7 classes in 34.053 seconds 
[13:24:37:485] com.gluonhq.higgs.Higgs: ERROR for class com.sun.javafx.property.adapter.PropertyDescriptor$Listener 
[13:24:37:487] com.gluonhq.higgs.Higgs: ERROR for class com.sun.javafx.property.adapter.ReadOnlyPropertyDescriptor$ReadOnlyListener 
[13:24:37:519] com.gluonhq.higgs.Higgs: ERROR for class com.sun.prism.j2d.J2DPrismGraphics$AdaptorPathIterator 
[13:24:37:520] com.gluonhq.higgs.Higgs: ERROR for class com.sun.prism.j2d.J2DPrismGraphics$AdaptorShape 
[13:24:37:522] com.gluonhq.higgs.Higgs: ERROR for class com.sun.prism.j2d.J2DPrismGraphics$FilterStroke 
[13:24:37:523] com.gluonhq.higgs.Higgs: ERROR for class com.sun.prism.j2d.paint.MultipleGradientPaint 
[13:24:37:525] com.gluonhq.higgs.Higgs: ERROR for class com.sun.prism.j2d.paint.MultipleGradientPaintContext 
[13:24:37:527] com.gluonhq.higgs.Higgs: ERROR for class com.sun.prism.j2d.print.J2DPrinterJob$J2DPageable 
[13:24:37:590] com.gluonhq.higgs.Higgs: non-fatal issue for class javafx.scene.web.JSObjectIosImpl (no known superclass) 

<truncated>

created
Error getting ProvisioninedDevices, ignore
[JVDBG] getProvisioningProfile asked, cached = null

<truncated>

provprofile asked, bid = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM and origbid = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM
provprofile asked, bid = com.gluonhq.samples.singleviewgvm.* and origbid = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM
provprofile asked, bid = com.gluonhq.samples.* and origbid = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM
provprofile asked, bid = com.gluonhq.* and origbid = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM
provprofile asked, bid = com.* and origbid = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM
provprofile asked, bid = * and origbid = com.gluonhq.samples.singleviewgvm.SingleViewWithGluonVM

<provisioning>

RES for signing = true
CODESIGN DONE
lockdown asked, answer = jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0020 size=8]
umbrella cp after lockdown = jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0020 size=8]
UploadInternal start
lockdown asked, answer = jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0028 size=8]
START WALKING filetree in uploadInternal
read from buffer: 11865
read from buffer: 569288
read from buffer: 1170
read from buffer: 112804

<truncated>

read from buffer: 119240
read from buffer: 47016
read from buffer: 11846
DONE WALKING filetree in uploadInternal
uploadInternal will now free client with pointer jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0038 size=8]
uploadInternal did now free client with pointer jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0038 size=8]
uploadInternal will now free lockdownpointer jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0028 size=8]
uploadInternal did now free lockdownpointer jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0028 size=8]
uploadInternal done
lockdown asked, answer = jnr.ffi.provider.BoundedMemoryIO[address=0x136eb7358 size=8]
free instProxyClientPointer at jnr.ffi.provider.BoundedMemoryIO[address=0x136eb7368 size=8]
freed instProxyClientPointer at jnr.ffi.provider.BoundedMemoryIO[address=0x136eb7368 size=8]
free lockdownclientpointer at jnr.ffi.provider.BoundedMemoryIO[address=0x136eb7358 size=8]
freed lockdownclientpointer at jnr.ffi.provider.BoundedMemoryIO[address=0x136eb7358 size=8]
umbrella cp will unlock = jnr.ffi.provider.BoundedMemoryIO[address=0x10e0f0020 size=8]
unlocked
launch
launch asked, wait on latch

launch Internal in thread
lockdown asked, answer = jnr.ffi.provider.BoundedMemoryIO[address=0x136ed4000 size=8]
ERROR! java.lang.IllegalArgumentException: result of lockdownd_start_service is -27
java.lang.IllegalArgumentException: result of lockdownd_start_service is -27
        at com.gluonhq.gvmbuild.MobileDeviceBridge.startService(MobileDeviceBridge.java:190)
        at com.gluonhq.gvmbuild.BosonAppBuilder$3.run(BosonAppBuilder.java:1045)​
```
I think, we are one step further. :smile:

**Comment by johanvos** [comment]
Thanks for that.
This time, the app should be installed on your phone, can you confirm that?
Also, the -27 value seems to be returned on "older" versions of iOS, are you on 11.2.5?

**Comment by bgmf** [comment]
Yes, I needed to double-check, but it was indeed installed (and I could start it). Because of the error I didn't bother to check the phone. :disappointed:
Also, the phone is not mine and I'm not "allowed" to install the latest version. So again: Yes, the installed version is an older one.

**Comment by kermiedefrog** [comment]
hey there, 
I have just started getting this -27 error when I moved to Gluon 5/Java9 combo ... app will upload to phone but then this error occurs ... 
I am running iOS 11.3 ... 

Is there something else that I needs to do like ???

**Comment by kermiedefrog** [comment]
I can answer my own question ... I can't confirm fully to what was wrong as I was doing a lot of stuff but there was an issue with an older app and the provisioning profile ... once I had fixed this up the newer G5/J9 apps on iOS started to work properly too ...

**Comment by johanvos** [comment]
ok, great to hear.
I think this issue can now be closed, unless someone disagrees?

**Comment by bgmf** [comment]
No, I think it's fine.
